Ura-558 add Variant AF and DP to cgppindel and remove small indels

Arun-bioinformatics opened the pull request:
- Converted into an app
- Updated to v1.1.0
- Included a python script which generates a annots.tsv table to include AF and DP data onto the FORMAT column onto cgppindel's vcf output.
- bcftools, tabix and bgzip needed to work
-  Address issue https://github.com/eastgenomics/eggd_cgppindel/issues/4
- This version filters out small variants (length <2bp).
- Changes “instanceType” from mem3_ssd1_v2_x4 to mem1_ssd2_v2_x4.
- Sets app to high priority.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)
<!-- Reviewable:end -->

Commit:
Python script to create tab index file for AF and DP
	new file:   tsv_file_generator.py

Commit:
Adding elements for AF and DP generation, need some help pls
	renamed:    tsv_file_generator.py -> resources/home/dnanexus/tsv_file_generator.py
	modified:   src/cgppindel.sh

Aisha-D reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 2 files reviewed, 4 unresolved discussions (waiting on @Arun-bioinformatics, @Jay-Miles, and @jethror1)

___
*[`resources/home/dnanexus/tsv_file_generator.py` line 27 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_ld73KmEy_7U4bxh:-NtW_ld73KmEy_7U4bxi:bcfn3l4) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L27)):*
> ```Python
>     parser.add_argument(
>         "-v",
>         "--vcfs",
> ```

vcfs -> vcfs
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 29 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_ZqN498izZ3J35XP:-NtW_ZqOAL_IaJMdXpDj:b-gt07qt) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L29)):*
> ```Python
>         "--vcfs",
>         type=str,
>         help="Panel filtered VCF(s) from which to generate excel workbook",
> ```

maybe better to say "cggpindel VCF"
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 39 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_qiE4JW77pCzLEMt:-NtW_qiE4JW77pCzLEMu:b-pikmkl) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L39)):*
> ```Python
> def read_vcf_df(input_vcf):
>     """
>     Reads vcf into pandas df, returns header as a list for output (bsvi) vcf
> ```

remove bsvi
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 51 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_v0T0llL0Pnf5HSz:-NtW_v0T0llL0Pnf5HT-:b7ymsdn) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L51)):*
> ```Python
> 
>     # read vcf records into df
>     vcf_df = pd.read_csv(input_vcf, sep="\t", comment="#", compression="infer",
> ```

do a sense check the number columns matches the number of elements in your cols list. This incase another VCF is inputted it recognises its does not have the expected number of columns


<!-- Sent from Reviewable.io -->

Aisha-D reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 2 files reviewed, 5 unresolved discussions (waiting on @Arun-bioinformatics, @Jay-Miles, and @jethror1)

___
*[`src/cgppindel.sh` line 68 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtWbVRBDHfReuyf0nh2:-NtWbVRBDHfReuyf0nh3:b-x5i9a0) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/src/cgppindel.sh#L68)):*
> ```Shell
>     tabix -s1 -b2 -e2 annots.tsv.gz
>     bcftools annotate -a annots.tsv.gz -h annots.hdr -c CHROM,POS,ID,REF,ALT,+FORMAT/AF,+FORMAT/DP \
>     out/cgppindel_output/*.flagged.vcf.gz > TODO:_I_do_not_know_what_to_write_here.vcf
> ```

also output this as an additional file to the amended VCF - changes to the dxjson is also needed https://github.com/eastgenomics/eggd_cgppindel/blob/URA-558_add_VAF_to_cgppindel/dxapp.json#L159 to add an extra output 


<!-- Sent from Reviewable.io -->

Commit:
Incorporated pysam for searching VCF format values

modified:   resources/home/dnanexus/tsv_file_generator.py
    Substituted read_vcf_df() for psam.VariantFile()
    generate_annotation_df() adapted to handle the VariantFile class

pep8speaks commented:
Hello @Arun-bioinformatics! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`resources/tsv_file_generator.py`](https://github.com/eastgenomics/eggd_cgppindel/blob/d23fc1cfc852d9654c7ff7e853339c2ae71b262f/resources/tsv_file_generator.py):

> [Line 41:1](https://github.com/eastgenomics/eggd_cgppindel/blob/d23fc1cfc852d9654c7ff7e853339c2ae71b262f/resources/tsv_file_generator.py#L41): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 82:22](https://github.com/eastgenomics/eggd_cgppindel/blob/d23fc1cfc852d9654c7ff7e853339c2ae71b262f/resources/tsv_file_generator.py#L82): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 83:22](https://github.com/eastgenomics/eggd_cgppindel/blob/d23fc1cfc852d9654c7ff7e853339c2ae71b262f/resources/tsv_file_generator.py#L83): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent



##### Comment last updated at 2024-05-10 10:47:49 UTC

Commit:
Minor space fixes
	modified:   resources/home/dnanexus/tsv_file_generator.py

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 104 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu-tU9W4WJDDid6P9_m:-Nu-tU9X2qb3LAoAlKxJ:b96vtag) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L104)):*
> ```Python
>     """
>     args = parse_args()
>     vcf_df = VariantFile(args.vcfs)
> ```

From parse_args(), it looks like args.vcfs can be a list of VCF files. However, VariantFile() operates on a single VCF. What happens if you pass it a list of multiple VCFs?

You might need to change this to be a loop which iterates over the contents of args.vcfs, creates a VariantFile() object for each, and deals with them sequentially.

As an aside, I'd be inclined to change the variable name to something like vcf_file instead of vcf_df; the variable is a VariantFile object which is not a dataframe, so the current variable name is misleading.


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 105 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu-vKEAER_JYD8QYY_L:-Nu-vKEBDzE2qxZqPnD7:b-qzyqzi) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L105)):*
> ```Python
>     args = parse_args()
>     vcf_df = VariantFile(args.vcfs)
>     annotation_data = generate_annotation_df(vcf_df)
> ```

In contrast, this variable actually IS a dataframe, so _df would be reasonable here. You don't need to change this one, just a comment.


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 89 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu02Ke49E2vwLlljx30:-Nu02Ke49E2vwLlljx31:b-yiix06) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L89)):*
> ```Python
>         # Generate vcf dataframe
>         data.append([variant.chrom, variant.pos, variant.id,
>                      variant.ref, variant.alts[0],
> ```

This is only taking the first element of the alts tuple. Do we not want to take the whole thing? Otherwise we're losing all alt alleles after the first.


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 54 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu045bn33C3bovvB7Se:-Nu045bn33C3bovvB7Sf:bhipstx) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L54)):*
> ```Python
>     Parameters
>     ----------
>     vcf_df : TYPE
> ```

Doesn't match the arg defined in the function name - I assume this should be vcf_file, which is a file and not a dataframe


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 59 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu03yeMAXMulmOP0CPx:-Nu03yeMAXMulmOP0CPy:b-20nsuv) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L59)):*
> ```Python
>     Returns
>     -------
>     annotation_df : pd.DataFrame
> ```

Also doesn't match the current function output, 'data', which is a list of lists.

I recommend you do convert 'data' into a Pandas dataframe within this function - it will then be much easier to write to CSV in main() using pd.to_csv().


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 108 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu05SWS5Y9STMjnNR_g:-Nu05SWS5Y9STMjnNR_h:bee5z7q) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L108)):*
> ```Python
>     with open('annots.tsv', 'w', newline="", encoding="utf-8") as f:
>         writer = csv.writer(f, delimiter='\t')
>         writer.writerows(annotation_data)
> ```

I would recommend making the output of generate_annotation_df() a pd.dataframe, and then using pd.to_csv() to generate this output file.


<!-- Sent from Reviewable.io -->

Jay-Miles reviewed (commented):
Much better for using pysam, thanks for making that change.

Have left a few comments, mostly small, but the thing about passing single vs. multiple VCFs needs consideration.

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-:-Nu062Hg3nSMd4IrHdPY:b-syg898)* status: 0 of 2 files reviewed, 11 unresolved discussions (waiting on @Arun-bioinformatics and @jethror1)



<!-- Sent from Reviewable.io -->

jethror1 reviewed (changes requested):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 2 files reviewed, 14 unresolved discussions (waiting on @Aisha-D, @Arun-bioinformatics, and @Jay-Miles)

___
*[`resources/home/dnanexus/tsv_file_generator.py` line 50 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0M97M7t-_YoXBpUdI:-Nu0M97M7t-_YoXBpUdJ:b-v5n5qm) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L50)):*
> ```Python
>    ----------------------------------------------------------------------------
>    chr2  |12345|699m| TA  |  G  |   0.666   |  0.8432   |    243    |    614
>    ----------------------------------------------------------------------------
> ```

update this now there's no dataframe 



_Code quote:_
```Python
    Creates an annotation df from the given vcf file, structure of the df
    should look like this:

   ----------------------------------------------------------------------------
   CHROM | POS | ID | REF | ALT | NORMAL AF | TUMOUR AF | NORMAL DP | TUMOUR DP
   ----------------------------------------------------------------------------
   chr1  |12345|450f| AGT | TCA |     0     |  0.9999   |     0     |    6
   ----------------------------------------------------------------------------
   chr1  |54321|219b| AC  | TCA |   0.53    |   0.64    |     1     |    90
   ----------------------------------------------------------------------------
   chr2  |12345|699m| TA  |  G  |   0.666   |  0.8432   |    243    |    614
   ----------------------------------------------------------------------------
```
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 87 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0KoLR9D8qudFmG_2G:-Nu0KoLS35tHVSAsFX3n:b-68c830) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L87)):*
> ```Python
>         tumour_dp = tumour['PR'] + tumour['NR']
> 
>         # Generate vcf dataframe
> ```

this is currently making just a list of the variant details, instead of writing back to a tsv then using it to annotate, I'd consider directly writing the new INFO annotation using pyvcf and avoiding the intermediary file.

Also as Jay said, here you're always indexing for the first alt, so if this vcf is not first decomposed you will skip other alts from multi allelic variants. Can we confirm if it is first decomposed or have a check here



_Code quote:_
```Python
# Generate vcf dataframe
```
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 108 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu05SWS5Y9STMjnNR_g:-Nu0MD2v1RKUn3zz0gkT:b-s0s5b3) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L108)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

I would recommend making the output of generate_annotation_df() a pd.dataframe, and then using pd.to_csv() to generate this output file.
</blockquote></details>

since none of the data is now being written into a dataframe, I think using `csv.writer()` is fine and means that the pandas whl package isn't needed
___
*[`src/cgppindel.sh` line 68 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtWbVRBDHfReuyf0nh2:-Nu0O4Io187nUZ3J-wMk:b-yi6hg7) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/src/cgppindel.sh#L68)):*
<details><summary><i>Previously, Aisha-D (Aisha) wrote…</i></summary><blockquote>

also output this as an additional file to the amended VCF - changes to the dxjson is also needed https://github.com/eastgenomics/eggd_cgppindel/blob/URA-558_add_VAF_to_cgppindel/dxapp.json#L159 to add an extra output
</blockquote></details>

instead of doing shell redirection here you can use `-o` to specify the output file for bcftools.

also instead of using a wildcard here to match to the one vcf, I would use something like `vcf=$(find . -type f -name "*.vcf" )`, then you can just specify `$vcf` as your input

if you want to get the basename of your vcf (i.e strip the `.vcf` and (optionally) `.gz` you can do the following :

```
$ echo $vcf
./test.vcf

$ basename $(basename $vcf .gz) .vcf
test
```

then you can add whatever suffix for output as you want
___
*[`src/cgppindel.sh` line 70 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0Q31w6KiAfg2OXXdl:-Nu0Q31w6KiAfg2OXXdm:b-5v7lkr) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/src/cgppindel.sh#L70)):*
> ```Shell
>     out/cgppindel_output/*.flagged.vcf.gz > TODO:_I_do_not_know_what_to_write_here.vcf
> 
>     #TODO:I don't know how to add the bash tools bgzip, tabix and bcftools in the assetDepends 
> ```

https://github.com/eastgenomics/eggd_optimised_filtering/blob/43564393e01d1841cf179ea7235dee95703ebe29/dxapp.json#L72



_Code quote:_
```Shell
TODO
```


<!-- Sent from Reviewable.io -->

Commit:
Fix variant.alts, minor update on cgppindel.sh

	modified:   resources/home/dnanexus/tsv_file_generator.py
	modified:   src/cgppindel.sh

Arun-bioinformatics reviewed (commented):
Yes, so the script will be using a single vcf file, since eggd_cgppindel currently is run once per sample, but I am not sure if this should be modified to use multiple instead. Let me know.

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-:-NuZUm_l5VbW1vvxY8TO:b-6vps18)* status: 0 of 2 files reviewed, 14 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`resources/home/dnanexus/tsv_file_generator.py` line 27 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_ld73KmEy_7U4bxh:-NuYnaOS5nHBZ-oEq8IP:b-hoj9qr) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L27)):*
<details><summary><i>Previously, Aisha-D (Aisha) wrote…</i></summary><blockquote>

vcfs -> vcfs
</blockquote></details>

I am not sure ow that this comment means?
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 29 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_ZqN498izZ3J35XP:-NuYneqzAJfDtzsa9lcc:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L29)):*
<details><summary><i>Previously, Aisha-D (Aisha) wrote…</i></summary><blockquote>

maybe better to say "cggpindel VCF"
</blockquote></details>

Done.
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 39 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_qiE4JW77pCzLEMt:-NuYnlv-9Nd7vTc3JZxd:b-pkm3nb) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L39)):*
<details><summary><i>Previously, Aisha-D (Aisha) wrote…</i></summary><blockquote>

remove bsvi
</blockquote></details>

function read_vcf() is gone
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 51 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtW_v0T0llL0Pnf5HSz:-NuYofRODiVg6v-L-prb:b-n4z7lb) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/resources/home/dnanexus/tsv_file_generator.py#L51)):*
<details><summary><i>Previously, Aisha-D (Aisha) wrote…</i></summary><blockquote>

do a sense check the number columns matches the number of elements in your cols list. This incase another VCF is inputted it recognises its does not have the expected number of columns
</blockquote></details>

Not sure, the script has changed quite a bit and I am not sure if this comment still applies
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 50 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0M97M7t-_YoXBpUdI:-NuYw8Qk5bmUQA5m-sI7:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L50)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

update this now there's no dataframe
</blockquote></details>

Done.
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 54 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu045bn33C3bovvB7Se:-NuYwAb01PzQHdSExoIk:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L54)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

Doesn't match the arg defined in the function name - I assume this should be vcf_file, which is a file and not a dataframe
</blockquote></details>

Done.
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 59 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu03yeMAXMulmOP0CPx:-NuYwCFT67Qonib3Kf4A:b-xpr1jd) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L59)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

Also doesn't match the current function output, 'data', which is a list of lists.

I recommend you do convert 'data' into a Pandas dataframe within this function - it will then be much easier to write to CSV in main() using pd.to_csv().
</blockquote></details>

Yes, I did try doing that, but it slowed the code by a almost a whole minute, so I chose to convert the list of lists into a tsv file without the step of converting it into a pandas df.
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 87 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0KoLR9D8qudFmG_2G:-NuYwWO17BL8pEL4qE_u:bjo5fcz) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L87)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

this is currently making just a list of the variant details, instead of writing back to a tsv then using it to annotate, I'd consider directly writing the new INFO annotation using pyvcf and avoiding the intermediary file.

Also as Jay said, here you're always indexing for the first alt, so if this vcf is not first decomposed you will skip other alts from multi allelic variants. Can we confirm if it is first decomposed or have a check here
</blockquote></details>

The current function makes use of a pysam.VariantFile, which hopefully makes it more robust, as it searches for a format id (e.g. PU) rather than its index. However this can reviewed when doing unittesting.
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 89 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu02Ke49E2vwLlljx30:-NuYww3l9DeOOVVXUnoD:b-n0ch1d) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L89)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

This is only taking the first element of the alts tuple. Do we not want to take the whole thing? Otherwise we're losing all alt alleles after the first.
</blockquote></details>

You are right, from my test vcf file, all variants have one ALT, however, I have changed this to ", ".join(variant.alts), in case there is more than one ALT for a particular variant.
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 104 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu-tU9W4WJDDid6P9_m:-NuZIBR4D-d5M8UUN7QN:b3v7t0l) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L104)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

From parse_args(), it looks like args.vcfs can be a list of VCF files. However, VariantFile() operates on a single VCF. What happens if you pass it a list of multiple VCFs?

You might need to change this to be a loop which iterates over the contents of args.vcfs, creates a VariantFile() object for each, and deals with them sequentially.

As an aside, I'd be inclined to change the variable name to something like vcf_file instead of vcf_df; the variable is a VariantFile object which is not a dataframe, so the current variable name is misleading.
</blockquote></details>

No problem, it has now been renamed to vcf_file
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 105 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu-vKEAER_JYD8QYY_L:-NuZIM9wA1UmY6Lrf1DW:b6fo785) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L105)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

In contrast, this variable actually IS a dataframe, so _df would be reasonable here. You don't need to change this one, just a comment.
</blockquote></details>

Like mentioned previously, the variable annotation_data will be a list of lists, would _df be appropriate still?
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 108 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu05SWS5Y9STMjnNR_g:-NuZIYRt0DaMlzbDADBX:b5z7v9h) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L108)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

since none of the data is now being written into a dataframe, I think using `csv.writer()` is fine and means that the pandas whl package isn't needed
</blockquote></details>

I have tried converting it to a pd.dataframe() and using pd.to_csv() and found out that the overall script worked slower, than when using csv.writer directly.
___
*[`src/cgppindel.sh` line 68 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtWbVRBDHfReuyf0nh2:-NuZNfvH6WBRoW0Hs3nY:b-8r5apm) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/src/cgppindel.sh#L68)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

instead of doing shell redirection here you can use `-o` to specify the output file for bcftools.

also instead of using a wildcard here to match to the one vcf, I would use something like `vcf=$(find . -type f -name "*.vcf" )`, then you can just specify `$vcf` as your input

if you want to get the basename of your vcf (i.e strip the `.vcf` and (optionally) `.gz` you can do the following :

```
$ echo $vcf
./test.vcf

$ basename $(basename $vcf .gz) .vcf
test
```

then you can add whatever suffix for output as you want
</blockquote></details>

should I not include dx next to find?
___
*[`src/cgppindel.sh` line 70 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0Q31w6KiAfg2OXXdl:-NuZUQRh9Kce5u0KNHPd:bkwt1rq) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/src/cgppindel.sh#L70)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

https://github.com/eastgenomics/eggd_optimised_filtering/blob/43564393e01d1841cf179ea7235dee95703ebe29/dxapp.json#L72
</blockquote></details>

From this example, I see that the assets used are from here, https://platform.dnanexus.com/projects/Fkb6Gkj433GVVvj73J7x8KbV/data/app_assets
But what if I need assets that are not from this link, should I then have to upload this on DNAnexus at this particular address?


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 59 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu03yeMAXMulmOP0CPx:-Nux7rUL7Wx-zd9FiGn2:b-1f7n4f) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L59)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Yes, I did try doing that, but it slowed the code by a almost a whole minute, so I chose to convert the list of lists into a tsv file without the step of converting it into a pandas df.
</blockquote></details>

Ok, that's fine. The name of the output variable does still need to be updated in the function header though please.


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
I think dealing with one VCF at once is probably best





<!-- Sent from Reviewable.io -->

Commit:
Minor change to doctring
	modified:   resources/home/dnanexus/tsv_file_generator.py

Commit:
Minor update function name
	modified:   resources/home/dnanexus/tsv_file_generator.py

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 2 files reviewed, 9 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`resources/home/dnanexus/tsv_file_generator.py` line 59 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu03yeMAXMulmOP0CPx:-Nv23xB-84TKzjItUz4j:bb6un1j) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L59)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

Ok, that's fine. The name of the output variable does still need to be updated in the function header though please.
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 59 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu03yeMAXMulmOP0CPx:-Nv24UcnD-lCCoFjqNGl:bjwrypn) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L59)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Done
</blockquote></details>

The returned variable is called data, not annotation_data


<!-- Sent from Reviewable.io -->

Commit:
Include assetDepends htslib
	modified:   dxapp.json

Commit:
Address eggd_cgppindel issue, update to v1.1.0
	modified:   dxapp.json
    https://github.com/eastgenomics/eggd_cgppindel/issues/4

Commit:
update output variable generate_annotation_data(vcf_file)
	modified:   resources/home/dnanexus/tsv_file_generator.py

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 3 files reviewed, 9 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`resources/home/dnanexus/tsv_file_generator.py` line 59 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu03yeMAXMulmOP0CPx:-Nv2NWFi-VkdFji64IbT:buvekkb) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L59)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

The returned variable is called data, not annotation_data
</blockquote></details>

Ah I see, got it


<!-- Sent from Reviewable.io -->

Commit:
Delete TODO comment

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 3 files reviewed, 9 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`src/cgppindel.sh` line 68 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtWbVRBDHfReuyf0nh2:-Nv7RjWmC9VUGgJ7SX2v:bvjldsw) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/src/cgppindel.sh#L68)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

should I not include dx next to find?
</blockquote></details>

Done, I think
___
*[`src/cgppindel.sh` line 70 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0Q31w6KiAfg2OXXdl:-Nv7RkqoCjLjlnKRGRok:b-896fjx) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/src/cgppindel.sh#L70)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

From this example, I see that the assets used are from here, https://platform.dnanexus.com/projects/Fkb6Gkj433GVVvj73J7x8KbV/data/app_assets
But what if I need assets that are not from this link, should I then have to upload this on DNAnexus at this particular address?
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`dxapp.json` line 5 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7WVsC2NbsbBGnKiuI:-Nv7WVsD3xzc9Bu1TVAr:bexj2ny) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/dxapp.json#L5)):*
> ```JSON
>   "title": "cgppindel_v1.1.0",
>   "summary": "cgppindel app for DNAnexus",
>   "dxapi": "1.1.0",
> ```

This is the DNAnexus API version, I think this should still be v1.0.0


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`dxapp.json` line 52 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7XE7738bw-ILVLck6:-Nv7XE7738bw-ILVLck7:b6ungrs) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/dxapp.json#L52)):*
> ```JSON
>         "*fai"
>       ],
>       "default": {"$dnanexus_link": "file-FyFGvzQ41zgBq7xf4Gy9Q28g"},
> ```

Why has the genome index file changed if the genome file hasn't changed?


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 89 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7Y6TA-Eo9d-S7wFlV:-Nv7Y6TA-Eo9d-S7wFlW:baye6q6) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/resources/home/dnanexus/tsv_file_generator.py#L89)):*
> ```Python
>         # Generate vcf dataframe
>         annotation_data.append([variant.chrom, variant.pos, variant.id,
>                      variant.ref, ", ".join(variant.alts),
> ```

Not sure whether a comma is the best separator to join on, since this is subsequently getting written to a comma-separated file. Maybe change ', ' to ';'


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`resources/home/dnanexus/tsv_file_generator.py` line 89 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7Y6TA-Eo9d-S7wFlV:-Nv7Ycd1--DCpOlT3I7i:b-qauej8) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/resources/home/dnanexus/tsv_file_generator.py#L89)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

Not sure whether a comma is the best separator to join on, since this is subsequently getting written to a comma-separated file. Maybe change ', ' to ';'
</blockquote></details>

It's also possible that there is a specific delimiter which needs to be used here so that alts can be split correctly later on in the pipeline - might be worth checking how alts get decomposed.


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`src/cgppindel.sh` line 64 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7ZdhkEpgUGKhFVlGK:-Nv7ZdhkEpgUGKhFVlGL:b-hb8xu3) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/src/cgppindel.sh#L64)):*
> ```Shell
> 
>     # Add Allele frequency (AF) and Read depth (DP) onto cgppindel output file
>     vcf=$(find . -type f -name "*.vcf.gz" )
> ```

This command will return all gzipped VCFs in the current directory. Are we certain that this will always be a single file? Is there a way we could make this more robust?


<!-- Sent from Reviewable.io -->

jethror1 reviewed (changes requested):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 3 files reviewed, 13 unresolved discussions (waiting on @Aisha-D, @Arun-bioinformatics, and @Jay-Miles)

___
*[`dxapp.json` line 5 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7WVsC2NbsbBGnKiuI:-NvaSILtDcBxH_Axq9FA:b-8fx2u7) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/dxapp.json#L5)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

This is the DNAnexus API version, I think this should still be v1.0.0
</blockquote></details>

:+1: 
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 50 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0M97M7t-_YoXBpUdI:-NvaSyg24GGX49AgvGy4:bklqbai) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L50)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

has it? it still looks to be there
___
*[`src/cgppindel.sh` line 68 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtWbVRBDHfReuyf0nh2:-NvaV7ZkD9E887xnD6pr:b-i0xhg3) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/src/cgppindel.sh#L68)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Done, I think
</blockquote></details>

nope since you're wanting to use the gnu find to find the file written to output and not one in DNAnexus
___
*[`src/cgppindel.sh` line 70 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0Q31w6KiAfg2OXXdl:-NvaV_wcDl24Nrq6Yieo:b-monzhw) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/src/cgppindel.sh#L70)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Done
</blockquote></details>

if you need further assets (i.e. different tools that need precompiling), these would need building and documenting as an asset and getting seperately signed off, then they would be added to 001_References, but I think you only need bcftools, right?
___
*[`src/cgppindel.sh` line 64 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7ZdhkEpgUGKhFVlGK:-NvaTqE04QVpw4xKHKBY:b-lsttwz) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/src/cgppindel.sh#L64)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

This command will return all gzipped VCFs in the current directory. Are we certain that this will always be a single file? Is there a way we could make this more robust?
</blockquote></details>

from the above running of pindel, the home directory is mounted in to the container, and then the output dir is set to `out/cgppindel_out` (which is the app array output of pindel files), then the modified vcf is added to `out/output_vcf`

There shouldn't be any other vcfs in the app environment (I assume), but setting `vcf=$(find out/cgppindel_out -type f -name "*.vcf.gz" )` would ensure only the vcf in the output dir is picked up in the case of some additional vcf being in the app


<!-- Sent from Reviewable.io -->

Commit:
Minor fixes
	modified:   dxapp.json
	modified:   resources/home/dnanexus/tsv_file_generator.py
	modified:   src/cgppindel.sh

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 3 files reviewed, 13 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`dxapp.json` line 52 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7XE7738bw-ILVLck6:-NvHU2JT-cEV57ExU0AH:bamfzje) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/dxapp.json#L52)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

Why has the genome index file changed if the genome file hasn't changed?
</blockquote></details>

https://github.com/eastgenomics/eggd_cgppindel/issues/4
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 50 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0M97M7t-_YoXBpUdI:-NvfqYsC3SruWP4mmgEo:b-9smzpr) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/resources/home/dnanexus/tsv_file_generator.py#L50)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

has it? it still looks to be there
</blockquote></details>

Ah, apologies, I just understood what you meant
___
*[`resources/home/dnanexus/tsv_file_generator.py` line 89 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7Y6TA-Eo9d-S7wFlV:-NvHv1va-L1G6qvERx99:b7ppams) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/resources/home/dnanexus/tsv_file_generator.py#L89)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

It's also possible that there is a specific delimiter which needs to be used here so that alts can be split correctly later on in the pipeline - might be worth checking how alts get decomposed.
</blockquote></details>

the output file of the python script is a tab-separated file. However when it comes to the alts separator is something that I am not really sure. I thought that the alts will be separated as a comma (like the same way as the example from https://samtools.github.io/hts-specs/VCFv4.2.pdf.

I also have not found any vcf file that has a variant that has more than 1 type of alt.
___
*[`src/cgppindel.sh` line 68 at r1](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NtWbVRBDHfReuyf0nh2:-NvfrcME9WY_-jV_BIKl:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/54fb354a944529cea054f3f2dd175ce6f5c3bc36/src/cgppindel.sh#L68)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

nope since you're wanting to use the gnu find to find the file written to output and not one in DNAnexus
</blockquote></details>

Done.
___
*[`src/cgppindel.sh` line 70 at r3](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nu0Q31w6KiAfg2OXXdl:-NvfqwBi2ct6KWx2bwj0:bxire6n) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/9ffe84a0834158ca19934f7f13fc5fa42bb9f769/src/cgppindel.sh#L70)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

if you need further assets (i.e. different tools that need precompiling), these would need building and documenting as an asset and getting seperately signed off, then they would be added to 001_References, but I think you only need bcftools, right?
</blockquote></details>

bcftools, bgzip and tabix, which are all in htslib
___
*[`src/cgppindel.sh` line 64 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7ZdhkEpgUGKhFVlGK:-Nvfr_eX66celRx-1Nxr:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/src/cgppindel.sh#L64)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

from the above running of pindel, the home directory is mounted in to the container, and then the output dir is set to `out/cgppindel_out` (which is the app array output of pindel files), then the modified vcf is added to `out/output_vcf`

There shouldn't be any other vcfs in the app environment (I assume), but setting `vcf=$(find out/cgppindel_out -type f -name "*.vcf.gz" )` would ensure only the vcf in the output dir is picked up in the case of some additional vcf being in the app
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 3 files reviewed, 13 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`dxapp.json` line 5 at r6](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-Nv7WVsC2NbsbBGnKiuI:-NvgGzZL1Zjllr26tN0l:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/0668621b195eb6a6cc76bd520144cbdb3aeac4d4/dxapp.json#L5)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

:+1:
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Commit:
Addition of a output_Spec key: output_vcf_with_vaf
	modified:   dxapp.json
	modified:   src/cgppindel.sh

Aisha-D reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 3 files reviewed, 11 unresolved discussions (waiting on @Arun-bioinformatics, @Jay-Miles, and @jethror1)

___
*[`dxapp.json` line 199 at r8](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NvkKY0M-tA6dOkHQUMw:-NvkKY0M-tA6dOkHQUMx:bbe7fct) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/17143aa664d1f05e7fce7ecec061a1cc8b1196c4/dxapp.json#L199)):*
> ```JSON
>       "name": "output_vcf_with_vaf",
>       "label": "array of cgppindel output files",
>       "class": "array:file",
> ```

as it is just the one file, keep it as `file` rather than `array:file`
___
*[`src/cgppindel.sh` line 77 at r8](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NvkNAD_1Tp4VUtdn6Gd:-NvkNAD_1Tp4VUtdn6Ge:bi4h5y9) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/17143aa664d1f05e7fce7ecec061a1cc8b1196c4/src/cgppindel.sh#L77)):*
> ```Shell
> 
>     # Move vcf file with VAF in speified folder
>     mv $basename.af.vcf.gz out/output_vcf_with_vaf
> ```

the directory doesnt exist, it'll need creation at the beginning of the file


<!-- Sent from Reviewable.io -->

jethror1 reviewed (commented):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 3 files reviewed, 9 unresolved discussions (waiting on @Aisha-D, @Arun-bioinformatics, and @Jay-Miles)



<!-- Sent from Reviewable.io -->

Commit:
Inclusion of annots.hdr, minor fixes in .sh and .json
	modified:   dxapp.json
	new file:   resources/home/dnanexus/annots.hdr
	modified:   src/cgppindel.sh

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 0 of 4 files reviewed, 9 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`dxapp.json` line 199 at r8](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NvkKY0M-tA6dOkHQUMw:-NvkPy_eA4y2rHdoS4Qd:bb6un1j) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/17143aa664d1f05e7fce7ecec061a1cc8b1196c4/dxapp.json#L199)):*
<details><summary><i>Previously, Aisha-D (Aisha) wrote…</i></summary><blockquote>

as it is just the one file, keep it as `file` rather than `array:file`
</blockquote></details>

Done
___
*[`src/cgppindel.sh` line 77 at r8](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NvkNAD_1Tp4VUtdn6Gd:-NvkXeb61stD47fFAzrX:bb6un1j) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/17143aa664d1f05e7fce7ecec061a1cc8b1196c4/src/cgppindel.sh#L77)):*
<details><summary><i>Previously, Aisha-D (Aisha) wrote…</i></summary><blockquote>

the directory doesnt exist, it'll need creation at the beginning of the file
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

Commit:
Fixes to make it work with dx run

Commit:
Incorporate step of removing 2bp variant
	modified:   src/cgppindel.sh

jethror1 reviewed (approved):
Happy with this - @Arun-bioinformatics are you going to add some unit tests in another PR?

> Reviewed 6 of 6 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-:-NxNmv7P1W6tNEAACJoe:bi59efx)* status: all files reviewed, 6 unresolved discussions (waiting on @Aisha-D and @Jay-Miles)



<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`dxapp.json` line 216 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS0FcYBlWgevzpQ5rB:-NxS0FcYBlWgevzpQ5rC:b-mmqg38) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/dxapp.json#L216)):*
> ```JSON
>     "distribution": "Ubuntu",
>     "release": "20.04",
>     "version": "0",
> ```

Should this be 1.1.0?


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`src/cgppindel.sh` line 4 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS15Ar4VfRQUfa9yZe:-NxS15As73kuW3RYTezl:b-qunxww) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/src/cgppindel.sh#L4)):*
> ```Shell
> # cgppindel 1.1.0
> 
> set -e -x -o pipefail
> ```

FYI you can just write this as -exo


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`src/cgppindel.sh` line 75 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS1VAq1V-PMT8_pATU:-NxS1VAq1V-PMT8_pATV:b-4djfjc) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/src/cgppindel.sh#L75)):*
> ```Shell
>     "$vcf" > "$basename.tmp.af.vcf"
> 
>     # Remove variants less than 2 bp
> ```

If this is looking for LEN>2, will it exclude variants which have LEN=2? Is this the intended behaviour?


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
So when speaking to Wook, I was told to be reasonable on how much testing is done (whether it is unittested or not). However, in the confluence page associated to this update, I did make a few tests to see if the app was working as expected:
https://cuhbioinformatics.atlassian.net/wiki/spaces/DV/pages/3190849540/Update+cgppindel+app+v1.1.0+Incorporate+eggd+vcf+handler+for+uranus+functions#Test-2%3A-The-AF-and-DP-values-align-with-the-expected-values-for-both-NORMAL-and-tumour-samples-PASS

If you still think that unit testing should be done, then I am happy to bring this up to Wook.

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-:-NxSCrsZ8Xd2uXoSf41O:b-t240gp)* status: all files reviewed, 3 unresolved discussions (waiting on @Aisha-D and @Jay-Miles)

___
*[`dxapp.json` line 216 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS0FcYBlWgevzpQ5rB:-NxSBoehEObBkNRVhrAH:bcj0ntt) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/dxapp.json#L216)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

Should this be 1.1.0?
</blockquote></details>

I am not sure, for example eggd_tso500 has been updated a couple of times, but the value for this key has always been 0
https://github.com/eastgenomics/eggd_tso500/blob/master/dxapp.json#L244
___
*[`src/cgppindel.sh` line 75 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS1VAq1V-PMT8_pATU:-NxSCSDNDarSlDdHdKbC:bvja78k) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/src/cgppindel.sh#L75)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

If this is looking for LEN>2, will it exclude variants which have LEN=2? Is this the intended behaviour?
</blockquote></details>

Yes, it will exclude variants where LEN=2, this is a behaviour which is equivalent to what we already have on production in the eggc_vcf_handler_for_uranus app:

https://github.com/eastgenomics/eggd_vcf_handler_for_uranus/blob/main/src/code.sh#L136


<!-- Sent from Reviewable.io -->

Commit:
Mem change, moving to app, exo

	modified:   dxapp.json
- "mem3_ssd1_v2_x4" -> "mem1_ssd2_v2_x4"

	modified:   src/cgppindel.sh
- set frequency of instance usage in logs to 30 seconds

Jay-Miles commented:
*[`dxapp.json` line 216 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS0FcYBlWgevzpQ5rB:-NxSIzt2-nhWGOCbEn4s:b-88rxex) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/dxapp.json#L216)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

I am not sure, for example eggd_tso500 has been updated a couple of times, but the value for this key has always been 0
https://github.com/eastgenomics/eggd_tso500/blob/master/dxapp.json#L244
</blockquote></details>

Oh my bad, didn't see this is nested within the runSpec. Resolved


<!-- Sent from Reviewable.io -->

Jay-Miles commented:
*[`src/cgppindel.sh` line 75 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS1VAq1V-PMT8_pATU:-NxSJHh9Cra5WWE6Diuh:bl9fxfp) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/src/cgppindel.sh#L75)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Yes, it will exclude variants where LEN=2, this is a behaviour which is equivalent to what we already have on production in the eggc_vcf_handler_for_uranus app:

https://github.com/eastgenomics/eggd_vcf_handler_for_uranus/blob/main/src/code.sh#L136
</blockquote></details>

Ok. Could you correct the comment line then please?


<!-- Sent from Reviewable.io -->

Commit:
Change app priority to high, name and comment changes

modified:   dxapp.json
modified:   src/cgppindel.sh

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: 4 of 6 files reviewed, 2 unresolved discussions (waiting on @Aisha-D, @Jay-Miles, and @jethror1)

___
*[`src/cgppindel.sh` line 75 at r10](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5#-NxS1VAq1V-PMT8_pATU:-NxWi2_RDxEwe3SpTHEA:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_cgppindel/blob/aff23876c6271abd787dda2b53cbd57ae4dc0ae8/src/cgppindel.sh#L75)):*
<details><summary><i>Previously, Jay-Miles wrote…</i></summary><blockquote>

Ok. Could you correct the comment line then please?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Aisha-D reviewed (approved):
> Reviewed 4 of 6 files at r10, 2 of 2 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: all files reviewed, 1 unresolved discussion (waiting on @Jay-Miles)



<!-- Sent from Reviewable.io -->

Jay-Miles reviewed (approved):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

Aisha-D reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_cgppindel/5)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->
